src/components/shared/Validations.tsx

```diff
@@ -1,90 +1,170 @@
-import { safeStringify } from '@stoplight/json';
-import { Dictionary } from '@stoplight/types';
-import { Popover } from '@stoplight/ui-kit';
+import { RegularNode } from '@stoplight/json-schema-tree';
+import { Flex, Text } from '@stoplight/mosaic';
+import { Dictionary, Primitive } from '@stoplight/types';
+import { capitalize, keys, omit, pick, pickBy, uniq } from 'lodash';
 import * as React from 'react';
 
-import { ViewModeContext } from '../../contexts';
-
 export interface IValidations {
-  required: boolean;
-  deprecated: boolean;
   validations: Dictionary<unknown>;
 }
 
-export const Validations: React.FunctionComponent<IValidations> = ({
-  required,
-  deprecated,
-  validations: { readOnly, writeOnly, ...validations },
-}) => {
-  const viewMode = React.useContext(ViewModeContext);
-  const validationCount = Object.keys(validations).length;
+const numberValidationNames = [
+  'minimum',
+  'maximum',
+  'minLength',
+  'maxLength',
+  'minItems',
+  'maxItems',
+  'exclusiveMinimum',
+  'exclusiveMaximum',
+];
 
-  const requiredElem = (
-    <div>
-      {required && <span className="ml-2 text-orange-7 dark:text-orange-6">required</span>}
-      {validationCount ? <span className="ml-1 text-darken-7 dark:text-lighten-6">+{validationCount}</span> : null}
-    </div>
-  );
+const excludedValidations = ['exclusiveMinimum', 'exclusiveMaximum', 'readOnly', 'writeOnly'];
 
-  // Show readOnly writeOnly validations only in standalone mode and only if just one of them is present
-  const showVisibilityValidations = viewMode === 'standalone' && !!readOnly !== !!writeOnly;
-  const visibility = showVisibilityValidations ? (
-    readOnly ? (
-      <span className="ml-2 text-darken-7 dark:text-lighten-6">read-only</span>
-    ) : (
-      <span className="ml-2 text-darken-7 dark:text-lighten-6">write-only</span>
-    )
-  ) : null;
+// this typing requires `numberValidationNames` to be defined `as const`.
+const numberValidationFormatters: Record<typeof numberValidationNames[number], (value: unknown) => string> = {
```

Comment:
⚠️  `typeof numberValidationNames[number]` only "works" if you declare
```tsx
const numberValidationNames = [
   // ...
] as const;
```
Otherwise it's a string. I mean, string is fine too if you prefer, but then let's express it as `string` rather than `typeof numberValidationNames[number]` 😄